types/redux-form/lib/selectors.d.ts

```diff
@@ -1,13 +1,14 @@
-import { FormErrors } from "redux-form";
+import { FormErrors, FormMeta } from "redux-form";
 
-export type DataSelector<FormData = {}, State = {}> = (formName: string) => (state: State) => FormData;
-export type ErrorSelector<FormData = {}, State = {}> = (formName: string) => (state: State) => FormErrors<FormData>;
-export type BooleanSelector<State = {}> = (formName: string) => (state: State) => boolean;
+export type DataSelector = <FormData = {}, State = {}>(formName: string) => (state: State) => FormData;
```

Comment:
Why is this an appropriate change? This means you're making each function generic which seems questionable.

Reply:
At the moment you cannot pass through the shape of the form data or redux state to the selectors which, I believe, means that typescript cannot then infer whether the keys you use are valid. The way it was before essentially made it always default to `{}` which then makes no sense in having the generics in the first place.

Also, in the definitions for v6 you could pass the generics through and since they way these selectors work has not changed I don't see why it would make sense to not allow the generics to be passed through, as before, if required especially as they are optional.

Hope this helps to clarify

Reply:
I _think_ I agree and I'd like to go ahead and merge this, but I'm still a bit confused. Could you add a test case, demonstrating intended usage, which would fail without your change?